GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -525,17 +528,26 @@ private void OnNewFileCreated(string relativePath, bool isDirectory)
                 {
                     if (isDirectory)
                     {
-                        GitCommandLineParser gitCommand = new GitCommandLineParser(this.Context.Repository.GVFSLock.GetLockedGitCommand());
-                        if (gitCommand.IsValidGitCommand)
-                        {
-                            // TODO(Mac): Ensure that when git creates a folder all files\folders within that folder are written to disk
-                            EventMetadata metadata = this.CreateEventMetadata(relativePath);
-                            metadata.Add("isDirectory", isDirectory);
-                            this.Context.Tracer.RelatedWarning(metadata, $"{nameof(this.OnNewFileCreated)}: Git created a folder, currently an unsupported scenario on Mac");
-                        }
-                        else
+                        if (!relativePath.Equals(GVFSConstants.DotGit.Root, StringComparison.OrdinalIgnoreCase))

```

Comment:
Isn't `relativePath` the name of the new file that was created? How could it be equal to `.git`?

Reply:
I don't think we need a new API. `PrjFS_ConvertDirectoryToVirtualizationRoot` could enumerate the root directory and mark everything that's already present as being in the root. Given that we're becoming more and more reliant on that flag being accurate, this seems like a good change to make.

Also this code as written is somewhat confusing, because we're inside an if block that just checked `if (!Virtualization.FileSystemCallbacks.IsPathInsideDotGit(relativePath))` - if we're not in `.git`, then how could it be `.git`? I get why, the `IsPathInsideDotGit` looks for `.git\`, not `.git`, but it ends up being hard to follow.

Reply:
Both good pieces of feedback, if the first suggestion works out I think we can stick with `IsPathInsideDotGit` being limited to checking `.git\` for now